apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -522,11 +522,12 @@ mod helpers {
                             to_schema,
                             &self.directive_deny_list,
                         );
-                        let (_, parsed) = JSONSelection::parse(&selection).map_err(|e| {
-                            FederationError::internal(format!(
-                                "could not parse fake selection for sibling field: {e}"
-                            ))
-                        })?;
+                        let (_, parsed) = JSONSelection::parse(&field_and_selection.sub_selection)
```

Comment:
you'll need a rebase to pick up the API change here — `parse` now returns a result.